### PR TITLE
Matomo Integration

### PR DESCRIPTION
**Pull request by tokdaniel**
- added next-matomo package for easier integration
- added useMatomo hook
- added track function with the possible tracking options.
- added an example for the tab-change in the index.tsx

there is some error with the rendering, react registers mamoto multiple times, which doesn't do anything, but I want to iron it out 

EDIT: the multiple render happens just in devmode, because of reactStrictMode, but not in production, so all is well

Screenshot 2022-09-28 at 20 54 34

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -12,7 +12,11 @@ const moduleExports = {
     // Warning: This allows production builds to successfully complete even if
     // your project has ESLint errors.
     ignoreDuringBuilds: false,
-  }
+  },
+  env: {
+    MATOMO_URL: process.env.MATOMO_URL,
+    MATOMO_SITE_ID: process.env.MATOMO_SITE_ID,
```
Comment (kasparkallas) [inline next.config.js:18]: This looks redundant, I don't think this adds anything at all.

Also if these are supposed to be visible to the browser then "NEXT_PUBLIC_" should be prefixed: https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser

File: src/hooks/useMatomo.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+  }, []);
+};
+
+type TrackerFunctionOptions = {
+  tracker: MatomoTracker;
+};
+
+type TrackerFunction = (
+  id: string,
```
Comment (kasparkallas) [inline src/hooks/useMatomo.ts:41]: We should probably type this if going forward.

**Review by kasparkallas — COMMENTED** [review]
The `track` & `register` combo looks clean for such a cross-cutting annoying concern! :+1:

File: next.config.js (hunk 1 above)
Comment (tokdaniel) [reply next.config.js:18]: if I didn't add them here, for some reason I haven't seen the env vars I'm not sure how they are parsed?

**Review by kasparkallas — APPROVED** [review]
Love it! Looked a bit into Matomo and it seems like a really reasonable web analytics platform that's aligned with our ethos.

**Pull request merged**